Indicators/OptionGreekIndicatorBase.cs

```diff
@@ -112,20 +112,122 @@ public abstract class OptionGreeksIndicatorBase : OptionIndicatorBase
         /// <param name="dividendYield">Dividend yield, as a constant</param>
-        /// <param name="period">The lookback period of historical volatility</param>
         /// <param name="optionModel">The option pricing model used to estimate the Greek</param>
         /// <param name="ivModel">The option pricing model used to estimate IV</param>
+        /// <param name="period">The lookback period of historical volatility</param>
         protected OptionGreeksIndicatorBase(string name, Symbol option, PyObject riskFreeRateModel, decimal dividendYield = 0.0m,
-            int period = 2, OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, OptionPricingModelType? ivModel = null)
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, OptionPricingModelType? ivModel = null, int period = 2)
             : this(name, option, RiskFreeInterestRateModelPythonWrapper.FromPyObject(riskFreeRateModel), 
-                new ConstantDividendYieldModel(dividendYield), period, optionModel, ivModel)
+                new ConstantDividendYieldModel(dividendYield), optionModel, ivModel, period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the OptionGreeksIndicatorBase class
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="option">The option to be tracked</param>
+        /// <param name="mirrorOption">The mirror option for parity calculation</param>
+        /// <param name="riskFreeRateModel">Risk-free rate model</param>
+        /// <param name="dividendYieldModel">Dividend yield model</param>
+        /// <param name="optionModel">The option pricing model used to estimate the Greek</param>
+        /// <param name="ivModel">The option pricing model used to estimate IV</param>
+        /// <param name="period">The lookback period of historical volatility</param>
+        protected OptionGreeksIndicatorBase(string name, Symbol option, Symbol mirrorOption, IRiskFreeInterestRateModel riskFreeRateModel, IDividendYieldModel dividendYieldModel,
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, OptionPricingModelType? ivModel = null, int period = 2)
+            : this(name, option, riskFreeRateModel, dividendYieldModel, optionModel, ivModel, period)
+        {
+            SetMirrorOptionContract(mirrorOption);
+            ImpliedVolatility.SetMirrorOptionContract(mirrorOption);
+            ImpliedVolatility.SetSmoothingFunction((iv, mirrorIV) =>
+            {
+                if (Strike > UnderlyingPrice && Right == OptionRight.Put)
+                {
+                    return mirrorIV;
+                }
+                else if (Strike < UnderlyingPrice && Right == OptionRight.Call)
+                {
+                    return mirrorIV;
+                }
+                return iv;
+            });
```

Comment:
Isn't this the default smoothing function? Don't think it's implementation should be duplicated here but in a centralized location